pkg/visibility/api/rest/pending_workloads_cq_test.go

```diff
@@ -298,15 +316,17 @@ func TestPendingWorkloadsInCQ(t *testing.T) {
 				manager.AddOrUpdateWorkload(w)
 			}
 
-			for _, cq := range tc.clusterQueues {
-				info, err := pendingWorkloadsInCqRest.Get(ctx, cq.Name, tc.queryParams)
+			for req, resp := range tc.wantResponse {
+				info, err := pendingWorkloadsInCqRest.Get(ctx, req.queueName, req.queryParams)
 				if err != nil {
-					t.Fatal(err)
-				}
-				pendingWorkloadsInfo := info.(*visibility.PendingWorkloadsSummary)
-
-				if diff := cmp.Diff(tc.wantPendingWorkloads, pendingWorkloadsInfo.Items, cmpopts.EquateEmpty()); diff != "" {
-					t.Errorf("Pending workloads differ: (-want,+got):\n%s", diff)
+					if diff := cmp.Diff(resp.wantErr.Error(), err.Error()); diff != "" {
```

Comment:
we don't compare strings https://github.com/kubernetes/community/blob/master/sig-scheduling/CONTRIBUTING.md#technical-and-style-guidelines

maybe you can have `wantErrMatch func(error) bool`, and you can pass things like `errors.IsNotFound`